Connection leak investigation

zhenlineo opened the pull request:
Investigating on connection leak in #141

Commit:
Connection leak

zhenlineo commented on Neo4j.Driver/Neo4j.Driver/Internal/Connector/TcpSocketClient.cs:
```diff
@@ -79,7 +79,7 @@ private Stream Stream
             _uri = uri;
             _useTls = useTlsEncryption;
-            await _client.ConnectAsync(uri.Host, uri.Port).ConfigureAwait(true);
+            await _client.ConnectAsync(uri.Host, uri.Port).ConfigureAwait(false);
```
@cskardon Hello, I am looking at this code that you added before, but I do not understand why this await has to go back to its calling context. Do you still remember why configAwait has to be `true` here? Can I change it to `false`?

cskardon replied:
Errrrr - if I'm totally honest with you I have no idea why that is set to `true` - I would have thought we'd have put it to `false`. There's no reason why we'd need to stay in the same SyncContext - I don't even know why we'd have done that in the first place. 

Does it crop up anywhere else in the codebase?

zhenlineo replied:
No, this is the only place to have this value to be true. So I guess I will just change it to false then.

Commit:
Moved the start of tls handshake into socket client async start method
Always close the stream and socket client when close to avoid stream or connection leak
